src/data/history.ts

```diff
@@ -417,28 +429,35 @@ export const computeHistory = (
       return;
     }
 
+    const domain = computeDomain(entityId);
+
     const currentState =
       entityId in hass.states ? hass.states[entityId] : undefined;
-    const stateWithUnitorStateClass =
+    const numericStateFromHistory =
       !currentState &&
-      stateInfo.find((state) => state.a && attributesHaveUnits(state.a));
+      stateInfo.find((state) => state.a && isNumericFromAttributes(state.a));
```

Comment:
We can move this in an else when `isNumeric` is false?

Reply:
Shouldnt we use the same checks here as we do below? So also use the domain and `isNumericSensorEntity`?

Reply:
I don't think we have device_class in the history.